[vanillapdf] Add new port

jurzik opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

jurzik committed:
[vanillapdf] Add new port

jurzik committed:
[vanillapdf] Reformat manifest

jurzik committed:
[vanillapdf] Change version-string to version

jurzik committed:
[vanillapdf] Version database

vicroms commented on ports/vanillapdf/portfile.cmake:
```diff
@@ -0,0 +1,42 @@
+# Vanilla.PDF vcpkg portfile
+# https://github.com/vanillapdf/vanillapdf
+#
+# Maintainer: Juraj Zikmund <jzikmund@vanillapdf.com>
+# License: Apache-2.0
+#
+# This portfile builds Vanilla.PDF, a cross-platform SDK for creating and modifying PDF documents.
+# It supports both static and shared builds via the standard CMake `BUILD_SHARED_LIBS` option.
+
```
```suggestion
```

This information is already present in `vcpkg.json`. This comment should be removed.

vicroms commented on ports/vanillapdf/usage:
```diff
@@ -0,0 +1,29 @@
+
+    find_package(vanillapdf CONFIG REQUIRED)
+    target_link_libraries(myapp PRIVATE vanillapdf::vanillapdf)
+
+Headers are located under the include directory:
+
+    #include <vanillapdf/document.h>
+    #include <vanillapdf/syntax/file.h>
+    #include <vanillapdf/semantics/document.h>
+    #include <vanillapdf/contents/content_stream.h>
+
+Features:
+
+- PDF parsing and manipulation
+- Digital signature support (via OpenSSL)
+- JPEG/JPEG2000 decompression
+- zlib stream decoding
+
+Optional features (must be enabled manually):
+
+- Skia rendering: `--feature=skia-support`
+- Benchmarks: `--feature=benchmarks`
+- Tests: `--feature=tests`
+
+Supports both static and shared builds depending on the selected triplet.
+
+License: Apache-2.0
+Homepage: https://github.com/vanillapdf/vanillapdf
```
```suggestion
```

We should keep the usage instructions to a minimum. Users can refer to the online documentation for all the extra instructions.

vicroms commented on ports/vanillapdf/vcpkg.json:
```diff
@@ -0,0 +1,44 @@
+        "benchmark"
+      ]
+    },
+    "skia-support": {
+      "description": "Enable Skia graphics engine integration for rendering",
+      "dependencies": [
+        "skia"
+      ]
+    },
```
I don't see a corresponding call to `vcpkg_check_features` for this feature. 
Is it the case where`vanillapdf` detects that `skia` is installed and automatically enables it?

jurzik replied:
The support for skia is still very experimental without any added value for the users, so I have removed it in the last commit. Should I also remove benchmarks and tests features if they cannot be enabled right now? It will be configurable in 2.2.0, but I did not want to backport it not to break anything - this building chain with dependencies is really sensitive.

vicroms replied:
Please do, it's better not to confuse users with features that don't toggle any actual behavior.

vicroms reviewed (changes requested):
Please make the requested changes. I'll be marking this PR as draft, when the changes are done please set it as ready for review again.

jurzik committed:
Update ports/vanillapdf/portfile.cmake

Co-authored-by: Victor Romero <romerosanchezv@gmail.com>

jurzik committed:
Update ports/vanillapdf/usage

Co-authored-by: Victor Romero <romerosanchezv@gmail.com>

jurzik committed:
Remove skia feature from manifest

skia support is an experimental option for vanillapdf.
Current version cannot really make any use of this,
so I am removing this from public vcpkg file.

jurzik committed:
[vanillapdf] Version database

vicroms commented on ports/vanillapdf/vcpkg.json:
```diff
@@ -0,0 +1,38 @@
+    },
+    "zlib"
+  ],
+  "features": {
+    "benchmarks": {
+      "description": "Enable performance benchmarking tools",
+      "dependencies": [
+        "benchmark"
+      ]
+    },
+    "tests": {
+      "description": "Enable unit and integration tests",
+      "dependencies": [
+        "gtest"
+      ]
+    }
```
Remove features with no effect on build.

jurzik committed:
Remove disabled features

jurzik committed:
[vanillapdf] Update database
